src/Microsoft.TemplateEngine.Edge/Settings/ComponentManager.cs

```diff
@@ -105,23 +105,26 @@ public ComponentManager(ISettingsLoader loader, SettingsStore userSettings)
         public IEnumerable<T> OfType<T>()
             where T : class, IIdentifiedComponent
         {
-            if (!_componentIdsByType.TryGetValue(typeof(T), out HashSet<Guid> ids))
+            lock (_componentIdToAssemblyQualifiedTypeName)
```

Comment:
`lock yield` probably not a good idea.
Random thought: will converting `_componentIdToAssemblyQualifiedTypeName` to `ConcurrentDictionary` help? 
If component with same ID is coming from different dlls this shouldn't be an issue, as in theory we can use any of them.

Reply:
I agree, I will switch to `ConcurrentDictionary`

Reply:
Problem with introducing `ConcurrentDictionary` is that we need to change many other collections like Hashsets also in other files... Hence keeping `lock` as is...